diff --git a/shared/exec/CMakeLists.txt b/shared/exec/CMakeLists.txt
index 2254459..e9cf1a0 100644
--- a/shared/exec/CMakeLists.txt
+++ b/shared/exec/CMakeLists.txt
@@ -21,7 +21,7 @@ include(../../packaging/cmake-inc/linking/CMakeLists.txt)
 include(../../packaging/cmake-inc/packaging/CMakeLists.txt)
 
 wintc_resolve_library(gdk-3.0      GDK)
-wintc_resolve_library(gio-2.0      GIO)
+wintc_resolve_library(gio-unix-2.0 GIO)
 wintc_resolve_library(glib-2.0     GLIB)
 wintc_resolve_library(gtk+-3.0     GTK3)
 wintc_resolve_library(wintc-comgtk WINTC_COMGTK)
