diff --git a/shell/run/CMakeLists.txt b/shell/run/CMakeLists.txt
index 3c0d065..5aafcbc 100644
--- a/shell/run/CMakeLists.txt
+++ b/shell/run/CMakeLists.txt
@@ -22,6 +22,7 @@ include(../../packaging/cmake-inc/packaging/CMakeLists.txt)
 
 wintc_resolve_library(gdk-3.0        GDK)
 wintc_resolve_library(glib-2.0       GLIB)
+wintc_resolve_library(gio-unix-2.0 GIO)
 wintc_resolve_library(gtk+-3.0       GTK3)
 wintc_resolve_library(wintc-comctl   WINTC_COMCTL)
 wintc_resolve_library(wintc-comgtk   WINTC_COMGTK)
@@ -59,6 +60,7 @@ target_include_directories(
     SYSTEM
     BEFORE
     PRIVATE ${GDK_INCLUDE_DIRS}
+    PRIVATE ${GIO_INCLUDE_DIRS}
     PRIVATE ${GLIB_INCLUDE_DIRS}
     PRIVATE ${GTK3_INCLUDE_DIRS}
     PRIVATE ${WINTC_COMCTL_INCLUDE_DIRS}
@@ -71,6 +73,7 @@ target_include_directories(
 target_link_directories(
     wintc-shell-run
     PRIVATE ${GDK_LIBRARY_DIRS}
+    PRIVATE ${GIO_LIBRARY_DIRS}
     PRIVATE ${GLIB_LIBRARY_DIRS}
     PRIVATE ${GTK3_LIBRARY_DIRS}
     PRIVATE ${WINTC_COMCTL_LIBRARY_DIRS}
@@ -83,6 +86,7 @@ target_link_directories(
 target_link_libraries(
     wintc-shell-run
     PRIVATE ${GDK_LIBRARIES}
+    PRIVATE ${GIO_LIBRARIES}
     PRIVATE ${GLIB_LIBRARIES}
     PRIVATE ${GTK3_LIBRARIES}
     PRIVATE ${WINTC_COMCTL_LIBRARIES}
