diff --git a/shared/shellext/CMakeLists.txt b/shared/shellext/CMakeLists.txt
index 08ea8a3..b3d3f6d 100644
--- a/shared/shellext/CMakeLists.txt
+++ b/shared/shellext/CMakeLists.txt
@@ -24,6 +24,7 @@ include(../../packaging/cmake-inc/packaging/CMakeLists.txt)
 include(../../packaging/cmake-inc/resources/CMakeLists.txt)
 
 wintc_resolve_library(glib-2.0       GLIB)
+wintc_resolve_library(gio-unix-2.0 GIO)
 wintc_resolve_library(gtk+-3.0       GTK3)
 wintc_resolve_library(wintc-comgtk   WINTC_COMGTK)
 wintc_resolve_library(wintc-exec     WINTC_EXEC)
@@ -68,6 +69,7 @@ target_include_directories(
     SYSTEM
     BEFORE
     PRIVATE ${GLIB_INCLUDE_DIRS}
+    PRIVATE ${GIO_INCLUDE_DIRS}
     PRIVATE ${GTK3_INCLUDE_DIRS}
     PRIVATE ${WINTC_COMGTK_INCLUDE_DIRS}
     PRIVATE ${WINTC_EXEC_INCLUDE_DIRS}
@@ -77,6 +79,7 @@ target_include_directories(
 target_link_directories(
     libwintc-shellext
     PRIVATE ${GLIB_LIBRARY_DIRS}
+    PRIVATE ${GIO_LIBRARY_DIRS}
     PRIVATE ${GTK3_LIBRARY_DIRS}
     PRIVATE ${WINTC_COMGTK_LIBRARY_DIRS}
     PRIVATE ${WINTC_EXEC_LIBRARY_DIRS}
@@ -86,6 +89,7 @@ target_link_directories(
 target_link_libraries(
     libwintc-shellext
     PRIVATE ${GLIB_LIBRARIES}
+    PRIVATE ${GIO_LIBRARIES}
     PRIVATE ${GTK3_LIBRARIES}
     PRIVATE ${WINTC_COMGTK_LIBRARIES}
     PRIVATE ${WINTC_EXEC_LIBRARIES}
