diff --git a/base/smss/CMakeLists.txt b/base/smss/CMakeLists.txt
index aab998e..6b042dc 100644
--- a/base/smss/CMakeLists.txt
+++ b/base/smss/CMakeLists.txt
@@ -23,6 +23,7 @@ include(../../packaging/cmake-inc/linking/CMakeLists.txt)
 include(../../packaging/cmake-inc/packaging/CMakeLists.txt)
 
 wintc_resolve_library(glib-2.0       GLIB)
+wintc_resolve_library(gio-unix-2.0   GIO)
 wintc_resolve_library(gtk+-3.0       GTK3)
 wintc_resolve_library(wintc-comctl   WINTC_COMCTL)
 wintc_resolve_library(wintc-comgtk   WINTC_COMGTK)
@@ -65,6 +66,7 @@ target_include_directories(
     SYSTEM
     BEFORE
     PRIVATE ${GLIB_INCLUDE_DIRS}
+    PRIVATE ${GIO_INCLUDE_DIRS}
     PRIVATE ${GTK3_INCLUDE_DIRS}
     PRIVATE ${WINTC_COMCTL_INCLUDE_DIRS}
     PRIVATE ${WINTC_COMGTK_INCLUDE_DIRS}
@@ -78,6 +80,7 @@ target_include_directories(
 target_link_directories(
     wintc-smss
     PRIVATE ${GLIB_LIBRARY_DIRS}
+    PRIVATE ${GIO_LIBRARY_DIRS}
     PRIVATE ${GTK3_LIBRARY_DIRS}
     PRIVATE ${WINTC_COMCTL_LIBRARY_DIRS}
     PRIVATE ${WINTC_COMGTK_LIBRARY_DIRS}
@@ -91,6 +94,7 @@ target_link_directories(
 target_link_libraries(
     wintc-smss
     PRIVATE ${GLIB_LIBRARIES}
+    PRIVATE ${GIO_LIBRARIES}
     PRIVATE ${GTK3_LIBRARIES}
     PRIVATE ${WINTC_COMCTL_LIBRARIES}
     PRIVATE ${WINTC_COMGTK_LIBRARIES}
